src/data/markdown/docs/02 javascript api/07 k6-experimental/04 tracing/02 Client.md

```diff
@@ -0,0 +1,94 @@
+
+The `Client` constructor accepts an [`Options`](/javascript-api/k6-experimental/tracing/options) object as its only parameter.
+
+## Example
+
+This example demonstrates how to instantiate a tracing client and use it to instrument HTTP calls with trace context headers. It also illustrates how the standard `http` module can still be used side-by-side with it to perform non-instrumented HTTP calls.
+
+<CodeGroup labels={["example-tracing-client.js"]} lineNumbers={[]} showCopyButton={[true]}>
+
+
+```javascript
+import http from "k6/http";
+import { check } from "k6";
+import tracing from "k6/experimental/tracing";
+
+// Explicitly instantiating a tracing client allows to distringuish
+// instrumented from non-instrumented HTTP calls, by keeping APIs separate.
+// It also allows for finer-grained configuration control, by letting
+// users changing the tracing configuration on the fly during their
+// script's execution.
+const instrumentedHTTP = new tracing.Client({
+ propagator: "w3c",
+});
+
+const testData = { name: "Bert" };
+
+export default () => {
+ // Using the tracing client instance, HTTP calls will have
+ // their trace context headers set.
+ let res = instrumentedHTTP.request("GET", "http://httpbin.org/get", null, {
+ headers: {
+ "X-Example-Header": "instrumented/request",
+ },
+ });
+ check(res, {
+ "status is 200": (r) => r.status === 200,
+ });
+
+ // The tracing client offers more flexibility over
+ // the `instrumentHTTP` function, as it leaves the
+ // imported standard http module untouched. Thus,
+ // one can still perform non-instrumented HTTP calls
+ // using it.
+ res = http.post("http://httpbin.org/post", JSON.stringify(testData), {
+ headers: { "X-Example-Header": "noninstrumented/post" },
+ });
+ check(res, {
+ "status is 200": (r) => r.status === 200,
+ });
+
+ res = instrumentedHTTP.del("http://httpbin.org/delete", null, {
+ headers: { "X-Example-Header": "instrumented/delete" },
+ });
+ check(res, {
+ "status is 200": (r) => r.status === 200,
+ });
+};
+```
+
+</CodeGroup>
+
+
+## HTTP module functions equivalents
+
+The `Client` class exposes the same API as the standard `http` module. Except for the `batch` method, which is absent from `Client`. The following table lists the `Client` methods which have an equivalent in the standard `http` module:
+
+| Method | HTTP equivalent | Description |
+| :---------------------------------------------- | :------------------------------------------------ | :----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `Client.del(url, [body], [params])` | [http.del](/javascript-api/k6-http/del)** | Performs an instrumented HTTP DELETE request. The method has the same prototype as the `http.del` function and should transparently act as a drop-in replacement for it. |
```

Comment:
What is the meaning of the `**` after the `http` methods? I was expecting some footnote, but didn't see any.

Reply:
Typo inherited from copy pasting 🤦🏻 Fixed 👍🏻